File: src/main/java/net/softbell/bsh/dto/request/IotActionDto.java
package net.softbell.bsh.dto.request;

import java.util.HashMap;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 액션정보 DTO
 */
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class IotActionDto
{
	private Long actionId;
    private boolean enableStatus;
    private String description;
    private HashMap<Long, IotActionItemDto> mapActionItem;
}

File: src/main/kotlin/net/softbell/bsh/dto/request/IotActionDto.kt
package net.softbell.bsh.dto.request

/**
 * @author : Bell(bell@softbell.net)
 * @description : 액션정보 DTO
 */
data class IotActionDto (
    var enableStatus: Boolean = false,
    var description: String,
    var mapActionItem: HashMap<Long, IotActionItemDto>? // TODO Not Null 대체방법 강구
)